frontend/src/components/DataSourceSelector/DataSourceSelectorToggle.tsx

```diff
@@ -0,0 +1,46 @@
+import classNames from 'classnames';
+import React, { FunctionComponent, useEffect } from 'react';
+import { Dropdown, DropdownProps } from 'semantic-ui-react';
+import { SelectEvent } from './DataSourceSelector';
+
+export type SelectedDatasource = 'non-frozen' | 'frozen' | 'all';
+
+interface ComponentProps {
+  className?: string;
+  onSelect: (selectedData: SelectedDatasource) => void;
+  defaultSource?: SelectedDatasource;
+}
+
+const DataSourceSelectorToggle: FunctionComponent<ComponentProps> = ({ ...props }) => {
+  useEffect(() => {
+    if (props.defaultSource && props.defaultSource.length) {
```

Comment:
You don't need length to determine if a string value is there or not. Strings are easily truthy or false ... so
```suggestion
    if (props.defaultSource) {
```

Reply:
Updated